daemons/gptp/linux/build/Makefile

```diff
@@ -30,9 +30,19 @@
 
 ALTERNATE_LINUX_INCPATH=$(HOME)/header/include/
 
+# Check to see if PTP cross-timestamping is supported in hardware
+PRECISE_TIME_TEST = "\#include <linux/ptp_clock.h>\\n\
+	\#ifdef PTP_SYS_OFFSET_PRECISE\\nint main(){return 0;}\\n\#endif"
+HAS_PRECISE_TIME = $(shell echo $(PRECISE_TIME_TEST) | gcc -xc - \
+	-I$(ALTERNATE_LINUX_INCPATH) -o /dev/null > /dev/null 2>&1 ; echo $$?)
+
 CFLAGS_G = -Wall -std=c++0x -g -I. -I../../common -I../src \
```

Comment:
Wow, this is very clever. I personally think it is overkill in response to my "where does PTP_SYS_OFFSET" come from. I would have been 100% ok with a comment block in the .cpp module before the first usage of PTP_SYS_OFFSET_PRECISE explaining that kernels with precise time support have PTP_SYS_OFFSET_PRECISE defined in the linux/ptp_clock.h header file. My 2c. Others may have different opinions.

Sorry I was slow responding to the followup post on this topic.